paramiko/file.py

```diff
@@ -95,34 +95,42 @@ def flush(self):
 
     if PY2:
 
-        def next(self):
+        def next(self, encoding="utf8", errors="strict"):
             """
             Returns the next line from the input, or raises
             ``StopIteration`` when EOF is hit.  Unlike Python file
             objects, it's okay to mix calls to `next` and `readline`.
 
+            :param str encoding:
+                encoding for file paths/names and content (default: utf8)
+            :param str errors:
+                python's default encoding error handling mode (default: strict)
             :raises: ``StopIteration`` -- when the end of the file is reached.
 
             :returns: a line (`str`) read from the file.
             """
-            line = self.readline()
+            line = self.readline(encoding=encoding, errors=errors)
             if not line:
                 raise StopIteration
             return line
 
     else:
 
-        def __next__(self):
+        def __next__(self, encoding="utf8", errors="strict"):
```

Comment:
The `__next__` method doesn't have arguments. It is unable to use the arguments because Python iterator directly invokes it.

Instead, I suggest:
 * Add `encoding` and `errors` parameters to `__init__` method and store to the instance attributes
 * Use the attributes here

Reply:
I see. I should have noticed that....

I'll try and see what to do best. Thanks.

Reply:
I fixed this and merged the newest commits from the main branch.

Reply:
I think `readline` should use `self.encoding` and `self.errors` and remove added arguments, because passing two parameters to `__init__`.

Reply:
Oops. I'll check again.

Reply:
Better now?